samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatHistoryController.cs

```diff
@@ -35,42 +36,48 @@ public class ChatHistoryController : ControllerBase
     /// <param name="logger">The logger.</param>
     /// <param name="chatSessionRepository">The chat session repository.</param>
     /// <param name="chatMessageRepository">The chat message repository.</param>
+    /// <param name="chatParticipantRepository">The chat participant repository.</param>
     /// <param name="promptsOptions">The prompts options.</param>
     public ChatHistoryController(
         ILogger<ChatHistoryController> logger,
         ChatSessionRepository chatSessionRepository,
         ChatMessageRepository chatMessageRepository,
+        ChatParticipantRepository chatParticipantRepository,
         IOptions<PromptsOptions> promptsOptions)
     {
         this._logger = logger;
         this._chatSessionRepository = chatSessionRepository;
         this._chatMessageRepository = chatMessageRepository;
+        this._chatParticipantRepository = chatParticipantRepository;
         this._promptOptions = promptsOptions.Value;
     }
 
     /// <summary>
     /// Create a new chat session and populate the session with the initial bot message.
+    /// The regex pattern that is used to match the user id will match the following format:
+    ///    - 2 period separated groups of one or more hyphen-delimited alphanumeric strings.
+    /// The pattern matches two GUIDs in canonical textual representation separated by a period.
     /// </summary>
-    /// <param name="chatParameters">Object that contains the parameters to create a new chat.</param>
+    /// <param name="userId">The user ID.</param>
+    /// <param name="chatParameter">Contains the title of the chat.</param>
     /// <returns>The HTTP action result.</returns>
     [HttpPost]
-    [Route("chatSession/create")]
+    [Route("chatSession/create/{userId:regex(([[a-z0-9]]+-)+[[a-z0-9]]+\\.([[a-z0-9]]+-)+[[a-z0-9]]+)}")]
     [ProducesResponseType(StatusCodes.Status201Created)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<IActionResult> CreateChatSessionAsync(
-        [FromBody] ChatSession chatParameters)
+    public async Task<IActionResult> CreateChatSessionAsync([FromRoute] string userId, [FromBody] ChatSession chatParameter)
     {
-        var userId = chatParameters.UserId;
-        var title = chatParameters.Title;
-
-        var newChat = new ChatSession(userId, title);
+        // Create a new chat session
+        var newChat = new ChatSession(chatParameter.Title);
```

Comment:
Why are we accepting a `ChatSession` in the body and throwing it all out except the title? Could we just have a title in the route instead? Or a simple JSON with the title in the body?

Reply:
+1, my vote is for the 2nd option (title in body) to be closer to REST.